src/realizations/catchment/LSTM_Realization.cpp

```diff
@@ -72,14 +39,33 @@ LSTM_Realization::LSTM_Realization(
  */
 double LSTM_Realization::get_response(time_step_t t_index, time_step_t t_delta_s) {
 
-    double precip = this->forcing.get_next_hourly_precipitation_meters_per_second();
-
-    AORC_data forcing_data = this->forcing.get_AORC_data();
-    int error = model->run(t_delta_s, forcing_data.DLWRF_surface_W_per_meters_squared, 
-                           forcing_data.PRES_surface_Pa, forcing_data.SPFH_2maboveground_kg_per_kg, 
-                           precip, forcing_data.DSWRF_surface_W_per_meters_squared, 
-                           forcing_data.TMP_2maboveground_K, forcing_data.UGRD_10maboveground_meters_per_second, 
-                           forcing_data.VGRD_10maboveground_meters_per_second);
+    //Checking the time step used is consistent with that provided in forcing data
+    time_t t_delta = this->forcing->record_duration();
+    if (t_delta != t_delta_s) {    //Checking the time step used is consistent with that provided in forcing data
+        throw std::invalid_argument("Getting response using insonsistent time step with provided forcing data");
+    }
+    //Negative t_index is not allowed
+    if (t_index < 0) {
+        throw std::invalid_argument("Getting response of negative time step in Tshirt C Realization is not allowed.");
+    }
+    time_t start_time = this->forcing->get_data_start_time();
+    time_t stop_time = this->forcing->get_data_stop_time();
+    time_t t_current = start_time + t_index * t_delta_s;
```

Comment:
is it guaranteed that data start time will align with simulation start time?  Also, there is no guarantee that `t_delta_s` is the SAME delta at each call to get response.

And I'm not sure I'm tracking time here correctly...

`t_index` should be the `current_time` relative to the simulation driver (or whatever is calling `get_response`.  `t_delta_s` is the amount of seconds the call wants to advance relative to `t_index`, or `current_time`.  Is this just trying to convert that into time into the forcing provider's time reference?

Reply:
as this code is for the generic provider the forcing time step could be anything, it does not necessarily match the model time step or is covenant multiple of the model time step. At least for all existing providers it is not a dynamic value.

Reply:
DP interface does support retrieving arbitrary time slices...however, this legacy code was already fixed to the forcing timestep by virtue of the fact that the old Forcing object kept the timestep and time state--that has shifted out of the DataProvider interface, but this didn't add any new restrictions.

Reply:
> Also, there is no guarantee that t_delta_s is the SAME delta at each call to get response.
Perhaps not via the interface, but the rest of the framework is currently locked at 3600. That's going to require way more complex changes than you'll find here to change.